File: templates/csharp/General/i_o/MetaInfo/MetaInfo.cs
using System;

public class MetaInfo
{
    public string Message { get; set; } = "";
    public bool On { get; set; }

    // Constructor 1
    public MetaInfo(bool show)
    {
        On = show;
    }

    // Constructor 2
    public MetaInfo(string msg)
    {
        On = true;
        Message = msg;
        Console.Write(this.ToString());
    }

    // Copy Constructor (C# doesn't have native copy constructors, but this mimics the logic)
    public MetaInfo(MetaInfo o)
    {
        Message = "Constructing Copy(MetaInfo)";
        On = o.On; // Explicitly copying state if needed for ToString
        Console.Write(this.ToString());
    }

    // Finalizer / Destructor
    ~MetaInfo()
    {
        Message = "Destructing(MetaInfo)";
        // Note: Console.Write inside a finalizer may not always execute if the app is shutting down
        Console.Write(this.ToString());
    }

    // Replaces the C++ ostream << operator overload
    public override string ToString()
    {
        if (On)
        {
            return Message + Environment.NewLine;
        }
        return string.Empty;
    }
}

File: templates/c/General/i_o/MetaInfo/MetaInfo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

typedef struct {
    char message[100];
    bool on;
} MetaInfo;

void printMetaInfo(const MetaInfo *obj) {
    if (obj->on) {
        printf("%s\n", obj->message);
    }
}

void initMetaInfoBool(MetaInfo *obj, bool show) {
    obj->on = show;
    strcpy(obj->message, ""); // Initialize with an empty string
}

void initMetaInfoMsg(MetaInfo *obj, const char *msg) {
    obj->on = true;
    strncpy(obj->message, msg, sizeof(obj->message) - 1);
    obj->message[sizeof(obj->message) - 1] = '\0'; // Ensure null-termination
    printMetaInfo(obj);
}

void initMetaInfoCopy(MetaInfo *obj, const MetaInfo *o) {
    obj->on = o->on; // Copying the state
    strcpy(obj->message, "Constructing Copy(MetaInfo)");
    printMetaInfo(obj);
}

void destroyMetaInfo(MetaInfo *obj) {
    strcpy(obj->message, "Destructing(MetaInfo)");
    printMetaInfo(obj);
}